### ISSUE TITLE
Revert server registration confirmation screen for 5.0.0

### ISSUE THREAD

**Issue by paramat** [minetest/minetest#8108]
See #7237 especially https://github.com/minetest/minetest/issues/7237#issuecomment-455047301
This screen annoys developers and it seems some players too due to the messy implementation.
It certainly needs a complete rework after release.
We are even considering a temporary setting to disable it for 5.0.0, even more mess.
I prefer we just revert the screen for 5.0.0 and do it properly afterwards, it's useful but not essential for 5.0.0.

**Comment by paramat** [comment]
I disagree with #258 . If you are joining a server for the first time and it is prompting you to enter a password, then it is obvious you are registering an account with that server, the entered password will of course be stored and required in future. I think the only issue here is the password confirmation to reduce the chance of mistyping and being locked out.

**Comment by rubenwardy** [comment]
I'm in favour removing it, it doesn't actually help the user experience at all

**Comment by paramat** [comment]
I'm volunteering to code this.
I also want to apologise to srifqi for changing my mind and proposing to undo your work.

**Comment by srifqi** [comment]
I agree, too.

**Comment by ClobberXD** [comment]
I support this too. It adds quite a bit of polish to MT, but the implementation makes it very inconvenient for singleplayers, testers, modders, devs, etc.

**Comment by SmallJoker** [comment]
My comment from https://github.com/minetest/minetest/issues/7237#issuecomment-381045762 is still valid. I disagree reverting the confirmation screen. It's however fine to skip it for local/hosted games over the GUI.

**Comment by jastevenson303** [comment]
Just a thought I had: What if the switch to turn the confirmation dialog on/off was a server setting?  For my servers I would want it off, but for other servers, operators might want it on.

**Comment by paramat** [comment]
Not the fault of the author, but the confirmation screen seems awkward and messy, it always gives me a bad impression. It has the password confirmation on a different screen to the initial password entry. It even appears for the server hoster when they create a world.
Because the confirmation screen has not been released into a stable release, if we release it in 5.0.0 it will be in for 1 release then changed again in 5.1.0.
It has some benefit but i think the problems outweigh. 5.1.0 is likely to follow soon after 5.0.0 so it's only a few extra months to wait.

**Comment by SmallJoker** [comment]
@paramat 
> It even appears for the server hoster when they create a world.

Hence, I suggest to add an exception for locally hosted games (or where the server owner name is the player name). This covers most of the development issues mentioned many times before.

@Ezhh 
> It doesn't help people understand how to register
> Mobile players have said they find it confusing.

Even though, if they try to login, it's explained to them. An explanation is better than leaving them uncertain.

> Testers (including devs) don't like it.

From what I've seen it's mainly due to that it shows up for admins in local games as well. Solvable without troubles. By the way, I like it for the reason to ensure the password is entered correctly changing it afterwards would then only be possible by an admin interaction.

> The original PR author agrees something better is needed.

Yes sure something better is needed. It's still not known what the `Connect` button exactly does. Add a tooltip: `Create or join with an existing account`, for example?

> To get to this, the confirmation needs to be removed.

And afterwards you'll need the very similar code again with new nickname and another password input field. 

I would like to keep the confirmation dialogue for 5.0.0, players will give further feedbacks on how to improve it more. In the meantime it would probably be best to add the host nickname exception above, and maybe a temporary setting to turn it off for those who entirely dislike this feature.

**Comment by jastevenson303** [comment]
I think the concern may be that mobile players aren't going to be able to read what it says anyway, because it's not going to render correctly for them.  It's just going to look like some error message, they're going to hit the back button, and that will be the end of it.

I'll just do a quick test... Nope, the selected topic doesn't exist?  Okay, first I have to find some .APK to test and then I will report back.

**Comment by Ezhh** [comment]
> Even though, if they try to login, it's explained to them. An explanation is better than leaving them uncertain.

The problem is that they could not work out how to make an account. If they had thought to type a name and password into the boxes (the thing needed before ever seeing the confirmation), they would have already solved the problem.

> Hence, I suggest to add an exception for locally hosted games (or where the server owner name is the player name). 

No, this isn't enough;  I don't want to be forced to use it when joining a server, because some things I test, I test on a server. If it doesn't get removed, why is a setting too much to ask for?

**Comment by Ezhh** [comment]
The server owner can make the choice that it's the player's responsibility and simply not reset passwords, which is why I'd prefer to let the player decide. On by default is fine if you do this. (I wouldn't mind making it a server specific approach if the current confirmation wasn't so bad.)

Maybe get some server owner opinions before deciding a server setting is a good approach.

Either way can we please get some, any, solution to this before 5.0 is released?

**Comment by Ryan-Nolan** [comment]
"how is this fair to players"  this has nothing todo with the players.... if a player does not like a feature or decision of a server then they will move on to another server... letting the server owner decide is the right thing todo because it is the server owner that has to deal with the incompetence of some players actions.

As i said in my previous reply it is not the engines job to facilitate the players its job is to facilitate the game devs and server owners... period.

further more.....i don't get what all the hate about this is "from a server owner perspective" its a one time thing per server for normal players. 

Anyhow I've made my point clear, I'm unsubscribing from this thread.

**Comment by Ezhh** [comment]
>  its job is to facilitate the game devs and server owners... 

Me being a game dev and sever owner. Enough said. This is why I say get opinions of those this will matter to.

**Comment by paramat** [comment]
> I think it should be left to server owners to decide.....
[...] should also be a server config option to define if its required on a per server basis.

I don't think it needs to be optional once properly implemented with separate 'connect' and 'register' buttons. Password confirmation on first join is obviously a good idea and should be mandatory, no one wants to typo their password and get locked out. I've never seen password confirmation optional on any website due to how important it is.
Being a serverside option is also the most complex way to implement it. The setting needs sending to the client on connect, which means the option cannot control anything done on the client before connection to the server.

What we're discussing is what to do for 5.0.0 before a proper implementation: Remove until done properly, or leave the current implementation in but optional (so that those annoyed by the current implementation can turn it off). I feel there's some confusion here between what to do now and what to do eventually, Ryan-Nolan which are you addressing?
Once implemented properly it doesn't need to be optional.
> I also think that it shouldn't be up to normal players to decide if this is on or off,

I agree, many will turn it off through laziness but then cause more password typos as a result (however it shouldn't be a server option).

Most of us know this is an engine, however in this case there is no need for an option in the eventual implementation.
> Look at it like this: if I as a server owner decide to force disable the confirmation.... how is this fair to players who WANT the added security? Some will, and are indeed more prone to making mistypes, so they need it. As a server owner I shouldn't be able to deny them that security step if it helps them

I agree, it shouldn't be a serverside option, but also not a clientside option.
>  But by making it a server setting, I'd have to do this, because I hate the confirmation and why should I put up with a thing I hate on my own server?

The confirmation page being annoying is only the current implementation, so i'm fine with it being clientside-optional for 5.0.0. But the annoying aspect won't be an issue in the final implementation so no option will be needed then.

Hmm Ryan-Nolan's gone for good unfortunately. Ryan-Nolan we are discussing a simple temporary solution for 5.0.0, either remove or add a simple clientside option, there's a PR for each and both PRs are fine. We are in feature freeze so implementing your suggestion is not possible before 5.0.0, and the proper implementation will be done after release anyway. An option is only being discussed for the temporary solution because only the current implementation needs an option.

So that was confusing =)
To clarify this: We have 2 simple PRs for what to do now, both are fine.

**Comment by paramat** [comment]
Actually, i've changed my opinon about this being optional, i think it should be a clientside option in the final implementation.

**Comment by ClobberXD** [comment]
I agree. Client-side (enabled by default) seems the best way to go. Those who don't want this extra step should be able to disable this if they want.

**Comment by SmallJoker** [comment]
@Ezhh 
> If it doesn't get removed, why is a setting too much to ask for?

It isn't. I'm against removing the formspec, but not against adding a setting. A client setting is totally fine by me, as long it's enabled by default.

> Either way can we please get some, any, solution to this before 5.0 is released?

Yes sure. #8102 is OK. Why not just merge that one?
